src/CommunityToolkit.Maui.Core/Essentials/SpeechToText/ISpeechToText.shared.cs

```diff
@@ -16,10 +16,54 @@ public interface ISpeechToText : IAsyncDisposable
 	/// <returns>Final recognition result</returns>
 	Task<SpeechToTextResult> ListenAsync(CultureInfo culture, IProgress<string>? recognitionResult, CancellationToken cancellationToken);
 
+	/// <summary>
+	/// Converts speech to text in real time.
+	/// </summary>
+	/// <param name="culture">Speak language</param>
+	/// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+	Task StartListeningAsync(CultureInfo culture, CancellationToken cancellationToken);
+
+	/// <summary>
+	/// Stop listening.
+	/// </summary>
+	/// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+	Task StopListeningAsync(CancellationToken cancellationToken);
+
+	/// <summary>
+	/// Converts speech to text in real time.
+	/// </summary>
+	event EventHandler<OnSpeechToTextRecognitionResultUpdated> RecognitionResultUpdated;
+
+	/// <summary>
+	/// Final recognition result.
+	/// </summary>
+	event EventHandler<OnSpeechToTextRecognitionResultCompleted> RecognitionResultCompleted;
```

Comment:
I'm trying to understand this callback, if I'm using async/await pattern to start/stop listening I would spect that one of those methods (StartListeningAsync or StopListeneningAsync) returns the string result instead of having to subscribe to an eventHandler. Is something that prevents those methods from returning the result?

Reply:
we already have a method ListenAsync, that returns result and intermediate result. So the idea of these methods just start and stop listening. we may later add Pause and resume for example